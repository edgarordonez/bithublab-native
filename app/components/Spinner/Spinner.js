// @flow

import React from 'react';
import { ActivityIndicator, StyleSheet, Text, View } from 'react-native';

export const Spinner = () => (
  <View style={styles.container}>
    <ActivityIndicator size="large" color="#2684ff" />
    <Text style={styles.text}>Searching repositories...</Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 10,
  },
  text: {
    fontFamily: 'Roboto',
    fontSize: 15,
    textAlign: 'center',
    color: '#8c929d',
    marginTop: 15,
  },
});
