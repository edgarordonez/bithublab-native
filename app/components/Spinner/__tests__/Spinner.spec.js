// @noflow

import React from 'react';
import TestRenderer from 'react-test-renderer';

import { Spinner } from '../Spinner';

describe('Spinner', () => {
  it('Renders without crashing', () => {
    const rendered = TestRenderer.create(<Spinner />);
    expect(rendered).toMatchSnapshot();
  });
});
