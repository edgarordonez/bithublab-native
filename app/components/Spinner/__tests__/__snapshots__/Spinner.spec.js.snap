// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#2684ff"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#8c929d",
        "fontFamily": "Roboto",
        "fontSize": 15,
        "marginTop": 15,
        "textAlign": "center",
      }
    }
  >
    Searching repositories...
  </Text>
</View>
`;
